homeassistant/components/qwikswitch/binary_sensor.py

```diff
@@ -13,8 +13,8 @@
 
 async def async_setup_platform(hass, _, add_entities, discovery_info=None):
     """Add binary sensor from the main Qwikswitch component."""
-    if discovery_info is None:
-        return
```

Comment:
This check is still necessary, because someone could add this to their configuration.yaml:

```yaml
binary_sensor:
  plaform: qwikswitch
```

(it's a legacy thing)

Reply:
ok. fixed it and added a test for this case